Source/Core/sampleTerrain.js

```diff
@@ -89,6 +89,10 @@ function doSampling(terrainProvider, level, positions) {
       tileRequest.level
     );
     var tilePromise = requestPromise
+      // Make sure to generate our mesh for each tile first
+      //  because some tiles actually require the mesh to interpolate heights
+      //  (eg: ArcGISTiledElevationTerrainProvider)
+      .then(createMeshCreatorFunction(tileRequest))
```

Comment:
This is essentially the fix I've added. Call `createMesh` on each tile as it comes back. However; as mentioned, this causes CWT terrain `interpolateHeight` to return undefined  - I haven't figured out why yet :)